test/acceptance/tests.py

```diff
@@ -228,17 +227,38 @@ def do_staff_assessment(self, number_to_assess=0, options_selected=OPTIONS_SELEC
                 will grade all available submissions.
         """
         self.staff_area_page.visit()
-        self.staff_area_page.expand_staff_grading_section()
+        self.staff_area_page.click_staff_toolbar_button("staff-grading")
+        # Get the counts before checking out a submission for assessment.
         start_numbers = self.staff_area_page.available_checked_out_numbers
+        # Check out a submission.
+        self.staff_area_page.expand_staff_grading_section()
+        # Checked out number should increase, ungraded decrease.
+        ungraded = start_numbers[0]-1
+        checked_out = start_numbers[1]+1
+        self.staff_area_page.verify_available_checked_out_numbers((ungraded, checked_out))
         assessed = 0
         while number_to_assess == 0 or assessed < number_to_assess:
             continue_after = False if number_to_assess-1 == assessed else True
+            if ungraded == 0:
+                continue_after = False
             self.staff_area_page.staff_assess(options_selected, continue_after)
             assessed += 1
-            new_numbers = (start_numbers[0]-assessed, start_numbers[1])
-            self.staff_area_page.verify_available_checked_out_numbers(new_numbers)
-            if not continue_after or not self.staff_area_page.submissions_available():
+            if not continue_after:
+                self.staff_area_page.verify_available_checked_out_numbers((ungraded, checked_out-1))
                 break
+            else:
+                ungraded -=1
+                self.staff_area_page.verify_available_checked_out_numbers((ungraded, checked_out))
+
+    def refresh_page(self):
+        """
+        Helper method that waits for "unsaved changes" warnings to clear before refereshing the page.
+        """
+        EmptyPromise(
```

Comment:
More handling of "unsaved changes" was added in my last PR. The clearing of the warnings is asynchronous, and the bok choy tests previously attempted to reload the page immediately after pressing the submit button(s).